vim-patch:7.4.{1163,1164,1167,1173,1178,1181,1228},8.1.0711

zeertzjq opened the pull request:
#### vim-patch:7.4.1163
Problem:    Expressions "0 + v:true" and "'' . v:true" cause an error.
Solution:   Return something sensible when using a special variable as a
            number or as a string. (suggested by Damien)
https://github.com/vim/vim/commit/17a13437c9414a8693369a97f3be2fc8ad48c12e

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim as removed it.


#### vim-patch:7.4.1164
Problem:    No tests for comparing special variables.  Error in jsondecode()
            not reported.  test_json does not work Japanse system.
Solution:   Set scriptencoding. (Ken Takata) Add a few more tests. Add error.
https://github.com/vim/vim/commit/6039c7f05376f0e470cf62bf2757e653aea357f3

Code is be N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.


#### vim-patch:7.4.1167
Problem:    No tests for "is" and "isnot" with the new variables.
Solution:   Add tests.
https://github.com/vim/vim/commit/04369229657f182d35b471eb8b38f273a4d9ef65

Comment out tests involving v:none as Nvim has removed it.


#### vim-patch:7.4.1173
Problem:    No test for new behavior of v:true et al.
Solution:   Add a test.
https://github.com/vim/vim/commit/65591001e405cbaaf9772c9375d0bb6049cf9a3a

Comment out tests involving v:none as Nvim has removed it.


#### vim-patch:7.4.1178
Problem:    empty() doesn't work for the new special variables.
Solution:   Make empty() work. (Damien)
https://github.com/vim/vim/commit/767d8c1a1ae762ecf47297c168b8c23caf05d30a

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.


#### vim-patch:7.4.1181
Problem:    free_tv() can't handle special variables. (Damien)
Solution:   Add the variable type.
https://github.com/vim/vim/commit/6650a694547eb744afa060ec62dd8270e99db9f2

Code is N/A. This only ports the tests.


#### vim-patch:7.4.1228
Problem:    copy() and deepcopy() fail with special variables. (Nikolai
            Pavlov)
Solution:   Make it work.  Add a test.
https://github.com/vim/vim/commit/155500077c80cdb5d9c63996000c011b66a676bf

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.


#### vim-patch:8.1.0711: test files still use function!

Problem:    Test files still use function!.
Solution:   Remove the exclamation mark.  Fix overwriting a function.
https://github.com/vim/vim/commit/1e1153600c0377472d62cc553173fe555ddcf5a7

Some of the changes were already applied previously.

seandewar reviewed (commented):
test_eval_func.vim change can be applied to legacy/eval_spec.lua instead (line 669) :+1:

zeertzjq committed:
vim-patch:7.4.1178

Problem:    empty() doesn't work for the new special variables.
Solution:   Make empty() work. (Damien)
https://github.com/vim/vim/commit/767d8c1a1ae762ecf47297c168b8c23caf05d30a

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:7.4.1164

Problem:    No tests for comparing special variables.  Error in jsondecode()
            not reported.  test_json does not work Japanse system.
Solution:   Set scriptencoding. (Ken Takata) Add a few more tests. Add error.
https://github.com/vim/vim/commit/6039c7f05376f0e470cf62bf2757e653aea357f3

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:7.4.1163

Problem:    Expressions "0 + v:true" and "'' . v:true" cause an error.
Solution:   Return something sensible when using a special variable as a
            number or as a string. (suggested by Damien)
https://github.com/vim/vim/commit/17a13437c9414a8693369a97f3be2fc8ad48c12e

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:7.4.1181

Problem:    free_tv() can't handle special variables. (Damien)
Solution:   Add the variable type.
https://github.com/vim/vim/commit/6650a694547eb744afa060ec62dd8270e99db9f2

Code is N/A. This only ports the tests.

zeertzjq committed:
vim-patch:7.4.1167

Problem:    No tests for "is" and "isnot" with the new variables.
Solution:   Add tests.
https://github.com/vim/vim/commit/04369229657f182d35b471eb8b38f273a4d9ef65

Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:7.4.1173

Problem:    No test for new behavior of v:true et al.
Solution:   Add a test.
https://github.com/vim/vim/commit/65591001e405cbaaf9772c9375d0bb6049cf9a3a

Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:7.4.1228

Problem:    copy() and deepcopy() fail with special variables. (Nikolai
            Pavlov)
Solution:   Make it work.  Add a test.  Closes vim/vim#614.
https://github.com/vim/vim/commit/155500077c80cdb5d9c63996000c011b66a676bf

Code is N/A. This only ports the tests.
Comment out tests involving v:none as Nvim has removed it.

zeertzjq committed:
vim-patch:8.1.0711: test files still use function!

Problem:    Test files still use function!.
Solution:   Remove the exclamation mark.  Fix overwriting a function.
https://github.com/vim/vim/commit/1e1153600c0377472d62cc553173fe555ddcf5a7

Some of the changes were already applied previously.

zeertzjq commented on src/nvim/testdir/test_vimscript.vim:
```diff
@@ -1156,6 +1156,83 @@ func Test_type()
+    " call assert_equal(v:none, eval(string(v:none)))
+    call assert_equal(v:null, eval(string(v:null)))
+
+
```
Thanks!

seandewar reviewed (approved):
LGTM. Those were the only issues I found. :+1:
